src/compass/utils/runconfig.py

```diff
@@ -48,22 +49,29 @@ def load_validate_yaml(yaml_path: str, workflow_name: str) -> dict:
         error_channel.log(err_str)
         raise ValueError(err_str)
 
+    # Determine run config type based on existence of newlines
+    run_config_is_txt = '\n' in yaml_runconfig
+
+    if not run_config_is_txt and not os.path.isfile(yaml_runconfig):
+        raise FileNotFoundError(f'Yaml file {yaml_runconfig} not found.')
+
     # load yaml file or string from command line
-    if os.path.isfile(yaml_path):
-        try:
-            data = yamale.make_data(yaml_path, parser='ruamel')
-        except yamale.YamaleError as yamale_err:
-            err_str = f'Yamale unable to load {workflow_name} runconfig yaml {yaml_path} for validation.'
-            error_channel.log(err_str)
-            raise yamale.YamaleError(err_str) from yamale_err
-    else:
-        raise FileNotFoundError(f'Yaml file {yaml_path} not found.')
+    try:
+        if run_config_is_txt:
+            data = yamale.make_data(content=yaml_runconfig,
+                                    parser='ruamel')
+        else:
+            data = yamale.make_data(yaml_runconfig, parser='ruamel')
+    except yamale.YamaleError as yamale_err:
+        err_str = f'Yamale unable to load {workflow_name} runconfig yaml {yaml_runconfig} for validation.'
+        error_channel.log(err_str)
+        raise yamale.YamaleError(err_str) from yamale_err
 
     # validate yaml file taken from command line
     try:
         yamale.validate(schema, data)
     except yamale.YamaleError as yamale_err:
-        err_str = f'Validation fail for {workflow_name} runconfig yaml {yaml_path}.'
+        err_str = f'Validation fail for {workflow_name} runconfig yaml {yaml_runconfig}.'
```

Comment:
Same suggestion as above. Maybe we can re-use `yaml_runconfig_info` in the comment above.